### PR TITLE
[refactor] move code from MetricsDataController to MetricsService

### PR DESCRIPTION
**Pull request by pwallk**
## What's changed?

 - [x] org.apache.hertzbeat.warehouse.controller.MetricsDataController #2427

## Checklist

- [x]  I have read the [Contributing Guide](https://hertzbeat.apache.org/docs/community/code_style_and_quality_guide)
- [x]  I have written the necessary doc or comment.
- [x]  I have added the necessary unit tests and all cases have passed.

## Add or update API

- [ ] I have added the necessary [e2e tests](https://github.com/apache/hertzbeat/tree/master/e2e) and all cases have passed.

### CODE REVIEW

File: warehouse/src/main/java/org/apache/hertzbeat/warehouse/service/impl/MetricsDataServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    @Override
+    public Boolean getWarehouseStorageServerStatus() {
+        return historyDataReader.isPresent() && historyDataReader.get().isServerAvailable();
+    }
+
+    @Override
+    public MetricsData getMetricsData(Long monitorId, String metrics) {
+        boolean available = realTimeDataReader.isServerAvailable();
+        if (!available) {
+            throw new CommonException("real time store not available");
```
Comment (Calvin979) [inline warehouse/src/main/java/org/apache/hertzbeat/warehouse/service/impl/MetricsDataServiceImpl.java:63]: Looks like this `CommonException` is not catched by `MetricsDataController`.

File: warehouse/src/main/java/org/apache/hertzbeat/warehouse/service/impl/MetricsDataServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+import org.apache.hertzbeat.common.entity.message.CollectRep;
+import org.apache.hertzbeat.common.support.exception.CommonException;
+import org.apache.hertzbeat.warehouse.service.MetricsDataService;
+import org.apache.hertzbeat.warehouse.store.history.HistoryDataReader;
+import org.apache.hertzbeat.warehouse.store.realtime.RealTimeDataReader;
+import org.springframework.stereotype.Service;
+
+/**
+ *
+ */
```
Comment (Calvin979) [inline warehouse/src/main/java/org/apache/hertzbeat/warehouse/service/impl/MetricsDataServiceImpl.java:41]: Could you help complete comment here?

Comment (pwallk) [reply warehouse/src/main/java/org/apache/hertzbeat/warehouse/service/impl/MetricsDataServiceImpl.java:41]: okay

File: warehouse/src/main/java/org/apache/hertzbeat/warehouse/service/impl/MetricsDataServiceImpl.java (hunk 1 above)
Comment (pwallk) [reply warehouse/src/main/java/org/apache/hertzbeat/warehouse/service/impl/MetricsDataServiceImpl.java:63]: `CommonException` is caught by the `GlobalExceptionHandler` in `Manager`, making sure it is the same as before, and throwing an exception forward when it is not available

**Review by Calvin979 — APPROVED** [review]
lgtm

**Pull request merged**